frontend/hooks/useBuilds.js

```diff
@@ -0,0 +1,44 @@
+import { useQuery, useQueryClient, useMutation } from '@tanstack/react-query';
+import api from '@util/federalistApi';
+
+// Assumes the order of builds is based on createdAt desc
+export function getLatestBuildByBranch(builds) {
+  const latestBranches = [];
+
+  return builds.map((build) => {
+    if (build.completedAt && !latestBranches.includes(build.branch)) {
+      latestBranches.push(build.branch);
+      return { ...build, latestForBranch: true };
+    }
+
+    return { ...build, latestForBranch: false };
+  });
+}
+
+export function useBuilds(id) {
```

Comment:
I'd rename this variable `siteId` for clarity (and it will bind the second variable of the mutation function which I think right now might not be needed)

Reply:
Updated